panic in existsmissedtickets

Trying to call `existmissedtickets` with wsrpc (I appreciate that I am passing  bad input, but I wouldn't expect bad input to cause a panic.)

```
src := []byte("a0911d93fb6347f5d8bca8f11f3535144f9d146f15598c2565fcfb98a462db71")
dst := make([]byte, hex.EncodedLen(len(src)))
hex.Encode(dst, src)
hashes := []string{string(dst)}
var existsMissed string
err = c.Call(c.ctx, "existsmissedtickets", &existsMissed, hashes)
```

dcrd panic

```
2020-05-28 10:36:42.704 [DBG] RPCS: Received command <existsmissedtickets> from 127.0.0.1:41540
panic: runtime error: index out of range [32] with length 32

goroutine 1683 [running]:
encoding/hex.Decode(0xc014c007a0, 0x20, 0x20, 0xc00ae17a00, 0x80, 0x80, 0xa7b360, 0x121c961, 0xc01ad525a0)
	/usr/lib/go-1.14/src/encoding/hex/hex.go:69 +0x22b
main.decodeHashes(0xc01aadb6c0, 0x1, 0x4, 0x121c961, 0xc000101500, 0x7f9fd6ac4560, 0x0, 0x65fcfb98a462db71)
	/home/jholdstock/code/decred/dcrd/rpcserver.go:1474 +0x13e
main.handleExistsMissedTickets(0xdeea00, 0xc0001300b0, 0xc014738b00, 0xa62d00, 0xc012e68e80, 0xa7b301, 0x121c961, 0x0, 0xc000504cf0)
	/home/jholdstock/code/decred/dcrd/rpcserver.go:1502 +0x6d
main.(*rpcServer).standardCmdResult(0xc014738b00, 0xdeea00, 0xc0001300b0, 0xc01573b6d0, 0x129da80, 0x121c900, 0x0, 0x0)
	/home/jholdstock/code/decred/dcrd/rpcserver.go:5071 +0x9f
main.(*wsClient).serviceRequest(0xc010c16800, 0xdeea00, 0xc0001300b0, 0xc01573b6d0)
	/home/jholdstock/code/decred/dcrd/rpcwebsocket.go:1940 +0x2b2
main.(*wsClient).inHandler.func1(0xc010c16800, 0xdeea00, 0xc0001300b0, 0xc01573b6d0)
	/home/jholdstock/code/decred/dcrd/rpcwebsocket.go:1639 +0x49
created by main.(*wsClient).inHandler
	/home/jholdstock/code/decred/dcrd/rpcwebsocket.go:1638 +0x274b
```

Also affects existlivetickets and existexpiredtickets.

--- a/rpcserver.go
+++ b/rpcserver.go
@@ -1471,8 +1471,11 @@ func handleExistsAddresses(_ context.Context, s *rpcServer, cmd interface{}) (in
 func decodeHashes(strs []string) ([]chainhash.Hash, error) {
 	hashes := make([]chainhash.Hash, len(strs))
 	for i, s := range strs {
-		l, err := hex.Decode(hashes[i][:], []byte(s))
-		if err != nil || l != 32 {
+		if len(s) != 2*chainhash.HashSize {
+			return nil, rpcDecodeHexError(s)
+		}
+		_, err := hex.Decode(hashes[i][:], []byte(s))
+		if err != nil {
 			return nil, rpcDecodeHexError(s)
 		}
 		// unreverse hash string bytes

